src/index.js

```diff
@@ -82,10 +81,90 @@ async function fetchAndParseBills() {
   // add filenames
   bills = bills.map(bill => ({
     ...bill,
-    filename: `${format(bill.date, 'YYYY-MM')}_o2.pdf`
+    filename: `${format(bill.date, 'YYYY-MM')}_o2.pdf`,
+    vendor: 'O2',
+    currency: '€',
+    metadata: {
+      importDate: new Date(),
+      version: 1
+    }
   }))
 
-  return bills
+  // now parse data in pdf files and save associated bill for each bill, to avoid save all the pdf
+  // files in memory
+  bills = await bluebird.mapSeries(bills, async bill => {
+    log('info', `parsing pdf file for ${bill.date} bill`)
+    const result = await findAndAddAmount(bill)
+    log('info', `got amount ${bill.amount}`)
+    log('info', `Now saving this bill to Cozy`)
+
+    if (result.amount === null) {
+      // if we could not find an amount in the PDF, we only save the file
+      await saveFiles([result], fields.folderPath, {
+        contentType: 'application/pdf'
+      })
+    } else {
+      await saveBills([result], fields.folderPath, {
+        identifiers: ['o2'],
+        contentType: 'application/pdf'
+      })
+    }
+    return result
+  })
+}
```

Comment:
I am sure node 8 handles Promise without using bluebird and of course `async/await` syntax.

You may simplify with something like that:
```js
const updatedBills = await bills.map(bill => ({
  ...bill,
  amount: await getAmount(bill)
})).map(bill => ({
  ...bill,
  contentType: 'application/pdf'
}))
```